tests/OmniSharp.Tests/UpdateBufferFilterFacts.cs

```diff
@@ -89,7 +89,6 @@ public UpdateBufferFilterFacts(ITestOutputHelper output)
                     filePath: "transient.cs");
 
                 var newSolution = host.Workspace.CurrentSolution.AddDocument(document);
-                host.Workspace.TryApplyChanges(newSolution);
```

Comment:
With this being removed, I'm not sure if the test is actually testing the behavior it's supposed to.

If you're new to Roslyn, then one thing to understand is that everything is immutable by default.  This allows for some really powerful features, but also is confusing if you're not familiar with it.  With the `TryApplyChanges` being removed then it's not actually adding the file the workspaces solution.

Reply:
Thanks for your time to explain this! 

IIRC, once I got that path issue out of the way, it looked to me problem was caused because WorkspaceChange events being raised async, which meant test is checking count of docs before BufferManager can get the memo it needs to get rid of transient docs, explaining where the 3 docs are coming from. I tried with a magic Thread.Sleep, and then it had only doc - which may be what should be a correct result after all! I just didn't want to jump to conclusions since test was working like this for such a long time.

If what I assume is correct, test should wait for event to be raised after this line (preferably without magic thread sleep 😆)

Reply:
I am not sure why there should be 2 docs with "transient.cs" file path after initial buffer update - unless those are the same file added to 2 different projects which exist in the test workspace. If real docuiment gets added, it should kill both copies, by my reading of BufferManager.WorkspaceChanged code.

Reply:
Without the `TryApplyChanges`, the updated solution will never get applied to the workspace.

Reply:
FWIW, I think this whole test needs rewritten. The fact that it's trying to write files to disk is a bit suspicious. I think it should be rewritten to use a proper test project like most of our other tests. Right now, we're trying to write the contents of "transient.cs" to disk when the project it belongs to doesn't even exist on disk.